[2025-10-04T13:37:47.114+0000] {local_task_job_runner.py:120} INFO - ::group::Pre task execution logs
[2025-10-04T13:37:47.129+0000] {taskinstance.py:2073} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: retail_mlops_data_pipeline.process_data_and_train_model scheduled__2025-10-03T00:00:00+00:00 [queued]>
[2025-10-04T13:37:47.133+0000] {taskinstance.py:2073} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: retail_mlops_data_pipeline.process_data_and_train_model scheduled__2025-10-03T00:00:00+00:00 [queued]>
[2025-10-04T13:37:47.133+0000] {taskinstance.py:2303} INFO - Starting attempt 1 of 2
[2025-10-04T13:37:47.147+0000] {taskinstance.py:2327} INFO - Executing <Task(PythonOperator): process_data_and_train_model> on 2025-10-03 00:00:00+00:00
[2025-10-04T13:37:47.155+0000] {standard_task_runner.py:63} INFO - Started process 3964 to run task
[2025-10-04T13:37:47.157+0000] {standard_task_runner.py:90} INFO - Running: ['airflow', 'tasks', 'run', 'retail_mlops_data_pipeline', 'process_data_and_train_model', 'scheduled__2025-10-03T00:00:00+00:00', '--job-id', '6', '--raw', '--subdir', 'DAGS_FOLDER/retail_mlops_data_pipeline.py', '--cfg-path', '/tmp/tmpih97wgpy']
[2025-10-04T13:37:47.157+0000] {standard_task_runner.py:91} INFO - Job 6: Subtask process_data_and_train_model
[2025-10-04T13:37:47.196+0000] {task_command.py:426} INFO - Running <TaskInstance: retail_mlops_data_pipeline.process_data_and_train_model scheduled__2025-10-03T00:00:00+00:00 [running]> on host ip-172-31-17-35.ec2.internal
[2025-10-04T13:37:47.431+0000] {taskinstance.py:2644} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='airflow' AIRFLOW_CTX_DAG_ID='retail_mlops_data_pipeline' AIRFLOW_CTX_TASK_ID='process_data_and_train_model' AIRFLOW_CTX_EXECUTION_DATE='2025-10-03T00:00:00+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='scheduled__2025-10-03T00:00:00+00:00'
[2025-10-04T13:37:47.431+0000] {taskinstance.py:430} INFO - ::endgroup::
[2025-10-04T13:37:47.431+0000] {taskinstance.py:441} INFO - ::group::Post task execution logs
[2025-10-04T13:37:47.432+0000] {taskinstance.py:2890} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/root/airflow_venv/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 465, in _execute_task
    result = _execute_callable(context=context, **execute_callable_kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/root/airflow_venv/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 432, in _execute_callable
    return execute_callable(context=context, **execute_callable_kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/root/airflow_venv/lib/python3.12/site-packages/airflow/models/baseoperator.py", line 400, in wrapper
    return func(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/root/airflow_venv/lib/python3.12/site-packages/airflow/operators/python.py", line 235, in execute
    return_value = self.execute_callable()
                   ^^^^^^^^^^^^^^^^^^^^^^^
  File "/root/airflow_venv/lib/python3.12/site-packages/airflow/operators/python.py", line 252, in execute_callable
    return self.python_callable(*self.op_args, **self.op_kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/root/airflow/dags/retail_mlops_data_pipeline.py", line 36, in process_and_train
    raise FileNotFoundError(f"Raw data not found at {FULL_FILE_PATH}. Please place the CSV here.")
FileNotFoundError: Raw data not found at /root/airflow/dags/data/processed/customer_shopping_data.csv. Please place the CSV here.
[2025-10-04T13:37:47.439+0000] {taskinstance.py:1205} INFO - Marking task as UP_FOR_RETRY. dag_id=retail_mlops_data_pipeline, task_id=process_data_and_train_model, execution_date=20251003T000000, start_date=20251004T133747, end_date=20251004T133747
[2025-10-04T13:37:47.453+0000] {standard_task_runner.py:110} ERROR - Failed to execute job 6 for task process_data_and_train_model (Raw data not found at /root/airflow/dags/data/processed/customer_shopping_data.csv. Please place the CSV here.; 3964)
[2025-10-04T13:37:47.491+0000] {local_task_job_runner.py:240} INFO - Task exited with return code 1
[2025-10-04T13:37:47.506+0000] {taskinstance.py:3482} INFO - 0 downstream tasks scheduled from follow-on schedule check
[2025-10-04T13:37:47.507+0000] {local_task_job_runner.py:222} INFO - ::endgroup::
